source/m2mtlvserializer.cpp

```diff
@@ -164,9 +177,44 @@ bool M2MTLVSerializer::serialize_multiple_resource(const M2MResource *resource,
     return success;
 }
 
+/* See, OMA-TS-LightweightM2M-V1_0-20170208-A, Appendix C,
+ * Data Types, Integer, TLV Format 
+ * 
+ * This is easy to change to support 8, 16 or 32 interger. */
+template <typename T> void M2MTLVSerializer::set_value_int(T value, uint8_t *buffer, uint32_t &size)
+{
+    size = sizeof(value);
+
+    /* write bytes to big endian order in buffer */
+    for (uint8_t ind=0;ind<size;ind++) {
+        buffer[size-ind-1] = ((value >> (ind*8)) & 0xFF);
+    }
+}
+
+template void M2MTLVSerializer::set_value_int<int8_t>(int8_t value, uint8_t *buffer, uint32_t &size);
```

Comment:
For unit test I need to copy/ make stub for common_functions.h for common_write_64_bit. 

I do not like way how common_write_64_bit has been implement and how it must use. It is working but I do not like it. Future plan is also to use automatic or manual sizing of integer. This will be decided later.  For different sizing I want to use my common loop against of common_write_64_bit, common_write_32_bit or  common_write_16_bit.